catalog/app/utils/validators.js

```diff
@@ -0,0 +1,15 @@
+const mkValidator = (name, test) => (v) => v && !test(v) ? name : undefined;
+
+const matches = (re) => (str) => re.test(str);
+
+export const required = (v) => v ? undefined : 'required';
+
+// taken from angular
+const EMAIL_RE =
+    /^(?=.{1,254}$)(?=.{1,64}@)[-!#$%&'*+/0-9=?A-Z^_`a-z{|}~]+(\.[-!#$%&'*+/0-9=?A-Z^_`a-z{|}~]+)*@[A-Za-z0-9]([A-Za-z0-9-]{0,61}[A-Za-z0-9])?(\.[A-Za-z0-9]([A-Za-z0-9-]{0,61}[A-Za-z0-9])?)*$/;
```

Comment:
That's definitely overkill - just do a basic sanity check.

Also, it's wrong - e.g., see https://en.wikipedia.org/wiki/Email_address#Examples

Reply:
it is a basic sanity check by the opinion of Angular authors). in the previous version the regexp was 3 times shorter